ShoppingCart-topic1/src/Counter.java

```diff
@@ -0,0 +1,20 @@
+public class Counter {

```

Comment:
mmm, maybe with some mechanism for processes that use shared variables (like the problem of critical section), or a boolean variable to indicate when you can use the counter